### PR TITLE
Add support for Python 3.12, drop 3.7

### PR DESCRIPTION
**Pull request by johningve**
Python 3.7 is EOL since June 2023.

This commit bumps pybind11 to version 2.11.1, which includes support for Python 3.12.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,15 +166,15 @@ Please visit [Zivid Knowledge Base][zivid-knowledge-base-url] for general inform
...
-| :--------------- | :------------------------ |
-| Ubuntu 23.10     | 3.11                      |
-| Ubuntu 23.04     | 3.11                      |
-| Ubuntu 22.04     | 3.10                      |
-| Ubuntu 20.04     | 3.8                       |
-| Fedora 38        | 3.11                      |
-| Fedora 37        | 3.11                      |
-| Windows 10       | 3.7, 3.8, 3.9, 3.10, 3.11 |
+| Operating System | Python version             |
+| :--------------- |:---------------------------|
```
Comment (nedrebo) [inline README.md:170]: Why? This makes the two collums different.

**Review by nedrebo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (johningve) [reply README.md:170]: Hmm, looks like my IDE reformatted this. Not sure why it made them different.

**Comment by eskaur** [comment]
Let me know when all the remaining jobs are green, and I can remove these from the "required checks" of the branch protection rules

image

File: README.md (hunk 1 above)
Comment (johningve) [reply README.md]: Reformatted the other column also.

**Review by eskaur — APPROVED** [review]

**Pull request merged**